[rpc] rename init_model_parallel to init_rpc

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#29762 [rpc] rename init_model_parallel to init_rpc**

Rename this API as discussed, since its use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

rohan-varma committed:
[distributed] rename init_model_parallel to init_rpc

Rename this API as discussed, since it's use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] rename init_model_parallel to init_rpc"


Rename this API as discussed, since it's use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

[ghstack-poisoned]

rohan-varma commented on torch/distributed/rpc/backend_registry.py:
```diff
@@ -54,7 +54,7 @@ def process_group_init_backend_handler(
     # Initialize ProcessGroup.
     if dist.is_initialized():
         raise RuntimeError(
-            "Default process group must not be initialized before `init_model_parallel`."
```
removed backticks

rohan-varma committed:
Update on "[rpc] rename init_model_parallel to init_rpc"


Rename this API as discussed, since its use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

[ghstack-poisoned]

pritamdamania87 commented on torch/distributed/rpc/__init__.py:
```diff
@@ -21,7 +21,7 @@ def is_available():
-    def init_model_parallel(
+    def init_rpc(
         self_name,
         backend=backend_registry.BackendType.PROCESS_GROUP,
```
We mention `model parallel` in the doc, should we rename that too?

rohan-varma replied:
Yep, instead of " Initializes model parallel primitives" I will change it to " Initializes rpc  primitives"

pritamdamania87 commented on test/rpc_test.py:
```diff
@@ -478,7 +478,7 @@ def test_sync_rpc(self):
     @dist_init(setup_model_parallel=False)
```
Should this flag be renamed too?

rohan-varma replied:
Good catch.

rohan-varma committed:
Update on "[rpc] rename init_model_parallel to init_rpc"


Rename this API as discussed, since its use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

[ghstack-poisoned]

zhaojuanmao reviewed (commented):
should we rename _init_rpc() as well?

rohan-varma commented:
> should we rename _init_rpc() as well?

I think we can rename it `_init_rpc_backend` since it sets up things like the process group backend.

pietern commented on torch/distributed/rpc/__init__.py:
```diff
@@ -31,7 +31,7 @@ def init_model_parallel(
     ):
         r"""
-        Initializes model parallel primitives such as the local rpc agent
+        Initializes rpc primitives such as the local rpc agent
```
Nit for later fixup: capitalize RPC.

rohan-varma committed:
Update on "[rpc] rename init_model_parallel to init_rpc"


Rename this API as discussed, since its use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[rpc] rename init_model_parallel to init_rpc"


Rename this API as discussed, since its use cases extend beyond only
model parallelism.

Differential Revision: [D18491743](https://our.internmc.facebook.com/intern/diff/D18491743/)

[ghstack-poisoned]
